components/test_raftstore/src/cluster.rs

```diff
@@ -522,13 +529,16 @@ impl<T: Simulator> Cluster<T> {
     pub fn shutdown(&mut self) {
         debug!("about to shutdown cluster");
         let keys;
-        match self.sim.try_read() {
+        match self.sim.read() {
             Ok(s) => keys = s.get_node_ids(),
-            Err(sync::TryLockError::Poisoned(e)) => {
-                let s = e.into_inner();
-                keys = s.get_node_ids();
+            Err(_) => {
+                if thread::panicking() {
```

Comment:
Why do you want to avoid the double panic? Do we recover from the initial panic elsewhere?

As far as I know for this sort of thing, the best practice is to have the destructor never panic and never return an error (Including from methods it calls). A shutdown method which can return an error or panic should only ever be called explicitly. I'm not sure if that works here.

Reply:
> Why do you want to avoid the double panic?

Double panic lead to coredump, which can ruin the test machine if core files are not handled correctly.

> Do we recover from the initial panic elsewhere?

Except some test cases that are expected to panic, TiKV never recovers panic.

> ...A shutdown method which can return an error or panic should only ever be called explicitly...

This method is only for tests. It's quite a common task to assert whether all threads exit without error. Returning error here will just make unwrap everywhere.